### ISSUE TITLE
Error validating payment XML file

### ISSUE THREAD

**Issue by Byloth** [OCA/bank-payment#515]
After the commit 7bf8d6aa8dfe7d738b55d6e77564d952afec2874, if you try to generate a payment XML file from `account.payment.order` Odoo will throw this error message:

```
The generated XML file is not valid against the official XML Schema Definition. The generated XML file and
the full error have been written in the server logs. Here is the error, which may give you an idea on the
cause of the problem : Element '{http://www.six-interbank-
clearing.com/de/pain.001.001.03.ch.02.xsd}PstlAdr': This element is not expected. Expected is ( 
{http://www.six-interbank-clearing.com/de/pain.001.001.03.ch.02.xsd}ClrSysMmbId )., line 40
```

---

This error is caused by this [few lines of codes](https://github.com/OCA/bank-payment/blob/7bf8d6aa8dfe7d738b55d6e77564d952afec2874/account_banking_pain_base/models/account_payment_order.py#L318,L326) of the function named `generate_fininst_postal_address`.

**Comment by sbidoul** [comment]
@Byloth thanks for the report.

The problem comes from https://github.com/OCA/l10n-switzerland/blob/379343c4ad40dcc31455513383b1d60077c5d6f3/l10n_ch_pain_credit_transfer/data/pain.001.001.03.ch.02.xsd#L577-L580 where the Swiss flavour forbids the PstlAdr element.

That is a strange practice to forbid an element that is allowed in the parent standard. But who am I to judge how the Swiss flavor is designed.

I think this should be addressed in l10n-switzerland with an override of the method that generates the postal address, in case the flavor is pain.001.001.03.ch.02.

Unfortunately I'll be offline for a short week, so I cannot work on this now.

**Comment by sbidoul** [comment]
https://github.com/OCA/bank-payment/commit/7bf8d6aa8dfe7d738b55d6e77564d952afec2874 is apparently causing issues with other pain flavors. I'll investigate deeper ASAP.

**Comment by sbidoul** [comment]
Closing as this is being handled in #517 and https://github.com/OCA/l10n-switzerland/pull/453

**Comment by Byloth** [comment]
Great @sbidoul,  
so is it functionally correct to handle this problem as I did in [`l10n-switzerland`'s PR #453](https://github.com/OCA/l10n-switzerland/pull/453)?

Has it been resolved in #517?  
Should I close the [PR](https://github.com/OCA/l10n-switzerland/pull/453) or should I keep it open?  
Could it be a problem with those changes made in #517?

**Comment by sbidoul** [comment]
@Byloth In https://github.com/OCA/bank-payment/pull/522 I provide the hook you need to resolve this properly in l10n-switzerland. So the best way is to override `has_fininst_postal_address` to return false for the swiss pain flavor.
